### PR TITLE
Add ticket search filters.

### PR DESCRIPTION
**Pull request by LautaroCesso**

### CODE REVIEW

File: client/src/app-components/ticket-list.js [hunk 1]
```diff
@@ -226,37 +282,6 @@ class TicketList extends React.Component {
         }
     }
 
-    compareFunction(row1, row2) {
```
Comment (mredigonda) [inline client/src/app-components/ticket-list.js]: Happy to see this go :')

File: client/src/reducers/search-filters-reducer.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        };
+    }
+
+    getTypeHandlers() {
+        return {
+            'SEARCH_FILTERS_ON_SUBMIT_FORM': this.onSubmitForm.bind(this),
+            'SEARCH_FILTERS_CHANGE_FORM': this.onFormChange,
+            'SEARCH_FILTERS_CHANGE_FILTERS': this.onFiltersChange.bind(this),
+            'SEARCH_FILTERS_SET_DEFAULT_FORM_VALUES': this.onSetDefaultFormValues.bind(this),
+            'SEARCH_FILTERS_CHANGE_SHOW_FILTERS': this.onChangeShowFilters.bind(this),
```
Comment (ivandiazwm) [inline client/src/reducers/search-filters-reducer.js:77]: please put declaration of this functions directly after` getTypeHandlers()`

File: client/src/actions/search-filters-actions.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+export default {
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/actions/search-filters-actions.js:1]: okey

File: client/src/app-components/ticket-list.js [hunk 4]
```diff
@@ -48,7 +49,9 @@ class TicketList extends React.Component {
         return (
             <div className="ticket-list">
                 <div className="ticket-list__filters">
-                    {(this.props.type === 'secondary' && this.props.showDepartmentDropdown) ? this.renderDepartmentsDropDown() : null}
+                    {(this.props.type === 'secondary' && this.props.showDepartmentDropdown) ?
+                        this.renderDepartmentsDropDown() :
+                        null}
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/app-components/ticket-list.js [hunk 5]
```diff
@@ -86,15 +94,21 @@ class TicketList extends React.Component {
...
             headers: this.getTableHeaders(),
             rows: this.getTableRows(),
             pageSize: 10,
-            comp: this.compareFunction,
-            page: this.props.page,
-            pages: this.props.pages,
-            onPageChange: this.props.onPageChange
+            page: page,
+            pages: pages,
+            onPageChange: onPageChange
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/actions/search-filters-actions.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    },
+    changeFilters(filters) {
+        return {
+            type: 'CHANGE_FILTERS',
+            payload: filters,
+        }
+    },
+    setDefaultFormValues() {
+        return {
+            type: 'SET_DEFAULT_FORM_VALUES',
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/actions/search-filters-actions.js]: okey

File: client/src/app-components/ticket-list.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/app-components/ticket-list.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/app-components/ticket-list.js [hunk 7]
```diff
@@ -161,13 +178,54 @@ class TicketList extends React.Component {
...
+                                {i18n('DATE')}
+                                {this.renderIcon('date')}
+                            </div>,
                     className: 'ticket-list__date col-md-2'
                 }
             ];
         }
     }
 
+    renderIcon(header) {
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/app-components/ticket-list.js [hunk 8]
```diff
@@ -161,13 +178,54 @@ class TicketList extends React.Component {
...
+        const {
+            orderBy,
+            showOrderArrows,
+            onChangeOrderBy
+        } = this.props;
+
+        if(showOrderArrows) {
+            return (
+                <Icon
+                    name={`arrow-${orderBy !== undefined ? this.getIconName(header, orderBy) : "down"}`}
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/app-components/ticket-list.js [hunk 9]
```diff
@@ -161,13 +178,54 @@ class TicketList extends React.Component {
...
+                    className="ticket-list__order-icon"
+                    color={orderBy ? this.getIconColor(header, orderBy) : "white"}
+                    onClick={() => onChangeOrderBy(header)} />
+            );
+        }
+    }
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/app-components/ticket-list.js [hunk 10]
```diff
@@ -161,13 +178,54 @@ class TicketList extends React.Component {
...
+
+    getIconName(header, orderBy) {
+        let name = "down";
+        orderBy = JSON.parse(orderBy);
+
+        if(orderBy.value === header) {
+            name = orderBy.asc === 0 ? "down" : "up";
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/app-components/ticket-list.js [hunk 11]
```diff
@@ -161,13 +178,54 @@ class TicketList extends React.Component {
...
+        }
+
+        return name;
+    }
+
+    getIconColor(header, orderBy) {
+        let color = "white";
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/app-components/ticket-list.js [hunk 12]
```diff
@@ -161,13 +178,54 @@ class TicketList extends React.Component {
...
+                    className="ticket-list__order-icon"
+                    color={orderBy ? this.getIconColor(header, orderBy) : "white"}
+                    onClick={() => onChangeOrderBy(header)} />
+            );
+        }
+    }
+
+    getIconName(header, orderBy) {
+        let name = "down";
+        orderBy = JSON.parse(orderBy);
```
[the remark this answers is not in the document]
Comment (LautaroCesso) [reply client/src/app-components/ticket-list.js]: okey

File: client/src/reducers/search-filters-reducer.js (hunk 2 above)
Comment (LautaroCesso) [reply client/src/reducers/search-filters-reducer.js:75]: okey

**Pull request closed without merging**